src/sonic-host-services/scripts/hostcfgd

```diff
@@ -49,13 +53,188 @@ def obfuscate(data):
+
+            self.update_feature_state(feature)
+            self.resync_feature_state(feature)
+
+    def update_feature_state(self, feature):
+        cached_feature = self._cached_config[feature.name]
+
+        if self.is_invariant_feature_state(cached_feature.state):
+            if cached_feature.state != feature.state:
+                syslog.syslog(syslog.LOG_INFO, "Feature {} service is {}".format(feature.name, feature.state))
+                self.resync_feature_state(cached_feature)
+            if feature.state == 'always_disabled':
+                self.disable_feature(feature)
+                syslog.syslog(syslog.LOG_INFO, "Feature {} is stopped and disabled".format(feature.name))
+            return
+
+        if feature.state == "enabled":
+            self.enable_feature(feature)
+            syslog.syslog(syslog.LOG_INFO, "Feature {} is enabled and started".format(feature.name))
+        elif feature.state == "disabled":
+            self.disable_feature(feature)
+            syslog.syslog(syslog.LOG_INFO, "Feature {} is stopped and disabled".format(feature.name))
+        else:
+            syslog.syslog(syslog.LOG_ERR, "Unexpected state value '{}' for feature {}"
+                          .format(feature.state, feature.name))
+
+    def update_feature_auto_restart(self, feature):
+        restart_config = "always" if feature.auto_restart == "enabled" else "no"
+        service_conf = "[Service]\nRestart={}\n".format(restart_config)
+        feature_names, feature_suffixes = self.get_feature_attribute(feature)
+
+        for service_name in feature_names:
+            dir_name = self.SYSTEMD_SERVICE_CONF_D.format(service_name)
+            if not os.path.exists(dir_name):
+                os.mkdir(dir_name)
+            with open(os.path.join(dir_name, 'auto_restart.conf'), 'w') as cfgfile:
+                cfgfile.write(service_conf)
+
+        try:
+            run_cmd("sudo systemctl daemon-reload", raise_exception=True)
+        except Exception as err:
+            syslog.syslog(syslog.LOG_ERR, "Feature '{}' failed to configure auto_restart".format(feature.name))
+            return
+
+    def get_feature_attribute(self, feature):
+        # Create feature name suffix depending feature is running in host or namespace or in both 
+        feature_names = (
+            ([feature.name] if feature.has_global_scope or not self.is_multi_npu else []) +
+            ([(feature.name + '@' + str(asic_inst)) for asic_inst in range(device_info.get_num_npus()) 
+                if feature.has_per_asic_scope and self.is_multi_npu])
+        )
+
+        if not feature_names:
+            syslog.syslog(syslog.LOG_ERR, "Feature '{}' service not available"
+                          .format(feature.name))
+
+        feature_suffixes = ["service"] + (["timer"] if feature.has_timer else [])
+
+        return feature_names, feature_suffixes
+
+    def is_invariant_feature_state(self, state):
```

Comment:
Is it necessary to create a separate function which just has one line code?

Reply:
is this really needed? what is the ROI of that?

Reply:
For readability purposes. Comparing:

```python
if cached_feature.state in ('always_enabled', 'always_disabled'):
```

vs

```python
if self.is_invariant_feature_state(cached_feature.state):
```

Reflects the purpose of the condition.